web/src/popover_menus_data.ts

```diff
@@ -266,7 +266,7 @@ export function get_topic_popover_content_context({
     const can_move_topic =
         !sub.is_archived && settings_data.user_can_move_messages_between_streams();
     const can_rename_topic =
-        !sub.is_archived && settings_data.user_can_move_messages_to_another_topic();
+        !sub.is_archived && stream_data.user_can_move_messages_within_channel(sub.stream_id);
```

Comment:
I feel like a future refactor might want to pull `sub.is_archived` part into the definition of `stream_data.user_can_move_messages_within_channel`, and similarly for the adjacent functions.